test/unit/foreman/templates/snippets/container_certs_setup_test.rb

```diff
@@ -0,0 +1,43 @@
+require 'test_helper'
+
+class ContainerCertsSetupTest < ActiveSupport::TestCase
+  def renderer
+    @renderer ||= Foreman::Renderer::SafeModeRenderer
+  end
+
+  def render_template(registration_host)
+    @snippet ||= File.read(Rails.root.join('app', 'views', 'unattended', 'provisioning_templates', 'snippet', 'container_certs_setup.erb'))
```

Comment:
Nit: on a [pathname](https://rubyapi.org/o/pathname) (which `Rails.root` is) you can call [read](https://rubyapi.org/o/pathname#method-i-read) directly
```suggestion
    @snippet ||= Rails.root.join('app', 'views', 'unattended', 'provisioning_templates', 'snippet', 'container_certs_setup.erb').read
```